### PR TITLE
pain system + addictions + traits

### PR DESCRIPTION
**Pull request by MilonPL**
## About the PR

adds the pain system
currently can be only picked with the chronic pain trait
adds chromatic aberration to your screen + popups
has two status effects `InPain` and `SuppressPain` to interact with it
adds the Addicted trait that gives you the Addicted component
minor changes to the addiction system to make it less shit (i burned it so hard)
makes substances use the proper systems now, addictine kill
completely mangles CD's files but I'm sure they won't mind (this is dual licensed under MIT anyways)

## Why / Balance

roleplay

## Technical details

massive shitcode
i lost any idea of what i'm doing 17 hours ago

## Media

image
image
image

## Requirements

- [x] I have tested all added content and changes.
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added Chronic pain and Addicted traits.
- tweak: Reworked the addictions system.

### CODE REVIEW

**Comment by HTMLSystem** [comment]
This seems really cool!! I love the idea of a chronic pain trait! Question though: Is the chromatic aberration constant or only when the pain flares up?

**Comment by MilonPL** [comment]
> This seems really cool!! I love the idea of a chronic pain trait! Question though: Is the chromatic aberration constant or only when the pain flares up?

Goes away when you're on painkillers (perhaps it should be the other way around? dunno)

**Comment by HTMLSystem** [comment]
Nah, i think it going away while on pain killers would be good. Either way I'm excited for this :)!!

**Comment by MilonPL** [comment]
@deltanedas you're going to love this one
i was high, drunk and/or sleep deprived while coding all this, and i have no idea how or why half of it works

File: Content.Server/DeltaV/Pain/PainSystem.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PainComponent, ComponentStartup>(OnInit);
```
Comment (deltanedas) [inline Content.Server/DeltaV/Pain/PainSystem.cs]: use MapInitEvent

**Comment by alterae** [comment]
massively, massively, MASSIVELY in favor of this to a degree that I cannot properly express

File: Content.Shared/DeltaV/Pain/SharedPainSystem.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public virtual void TrySuppressPain(EntityUid uid, float duration, PainSuppressionLevel level = PainSuppressionLevel.Normal)
+    {
+        if (!TryComp<PainComponent>(uid, out var comp))
+            return;
+
+        var ent = new Entity<PainComponent>(uid, comp);
+
+        UpdatePainSuppression(ent, duration, level);
```
Comment (deltanedas) [inline Content.Shared/DeltaV/Pain/SharedPainSystem.cs]: just pass `(uid, comp)` dont need new Entity

File: Content.Shared/DeltaV/Pain/SharedPainSystem.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+}
+
+// Used by the StatusEffect
+public enum PainSuppressionLevel
```
Comment (deltanedas) [inline Content.Shared/DeltaV/Pain/SharedPainSystem.cs]: make it extend byte

File: Resources/Prototypes/Reagents/narcotics.yml [hunk 4]
```diff
@@ -294,21 +282,17 @@
         type: Add
         time: 5
         refresh: false
-      - !type:ChemRerollPsionic #Nyano - Summary: lets the imbiber become psionic. 
+      - !type:ChemRerollPsionic #Nyano - Summary: lets the imbiber become psionic.
```
Comment (deltanedas) [inline Resources/Prototypes/Reagents/narcotics.yml:285]: bad line changed

File: Resources/Prototypes/_CD/Reagents/medicine.yml [hunk 5]
```diff
@@ -585,7 +477,7 @@
       - !type:SatiateHunger # causes digestion inefficiency during overdose
         factor: -1.2
         conditions:
-          - !type:ReagentThreshold 
+          - !type:ReagentThreshold
```
Comment (deltanedas) [inline Resources/Prototypes/_CD/Reagents/medicine.yml]: line

File: Resources/Prototypes/_CD/Reagents/medicine.yml [hunk 6]
```diff
@@ -596,7 +488,7 @@
       - !type:SatiateHunger # causes digestion inefficiency if you drink alcohol
         factor: -2.0
         conditions:
-          - !type:ReagentThreshold 
+          - !type:ReagentThreshold
```
Comment (deltanedas) [inline Resources/Prototypes/_CD/Reagents/medicine.yml]: line

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Review by deltanedas — APPROVED** [review]

**Pull request merged**